debugd/internal/cdbg/cmd/deploy.go

```diff
@@ -164,11 +173,14 @@ type deployOnEndpointInput struct {
 func deployOnEndpoint(ctx context.Context, in deployOnEndpointInput) error {
 	in.log.Infof("Deploying on %v", in.debugdEndpoint)
 
-	client, closer, err := newDebugdClient(ctx, in.debugdEndpoint)
+	client, conn, err := newDebugdClient(ctx, in.debugdEndpoint, in.log)
 	if err != nil {
 		return fmt.Errorf("creating debugd client: %w", err)
 	}
-	defer closer.Close()
+	defer conn.Close()
+	var wg sync.WaitGroup
+	defer wg.Wait()
```

Comment:
Maybe move this into the constructor to hide it from the caller and return a "closeAndWaiter" func instead. Not directly obvious if the order of calls matter here...

Reply:
i thought of hiding the WaitGroup too,  but it was equally handled in maltes code and is more extendable, since the routine could be part of a bigger worker group. 

Made the change in a commit.
How does it solve the problem of call order confusion though?